gradle-conjure/src/main/java/com/palantir/gradle/conjure/ConjurePlugin.java

```diff
@@ -136,246 +138,134 @@ public void apply(Project project) {
+                    String.format("Cannot enable '%s' without '%s'", projectName, objectsProjectName));
         }
 
-        project.project(derivedProjectPath(project, retrofitProjectName), subproj -> {
+        String upperSuffix = getUppercaseSuffix(projectSuffix);
+        return parentProject.project(derivedProjectPath(parentProject, projectName), subproj -> {
             subproj.getPluginManager().apply(JavaLibraryPlugin.class);
-
             ignoreFromCheckUnusedDependencies(subproj);
             addGeneratedToMainSourceSet(subproj);
-            project.getTasks().create("compileConjureRetrofit", ConjureGeneratorTask.class, task -> {
+            parentProject.getTasks().create("compileConjure" + upperSuffix, ConjureGeneratorTask.class, task -> {
                 task.setDescription(
-                        "Generates Retrofit interfaces for use on the client-side from your Conjure definitions.");
+                        String.format("Generates %s interfaces from your Conjure definitions.", upperSuffix));
                 task.setGroup(TASK_GROUP);
                 task.setExecutablePath(extractJavaTask::getExecutable);
-                task.setOptions(() -> optionsSupplier.get().addFlag("retrofit"));
+                task.setOptions(() -> optionsSupplier.get().addFlag(projectSuffix));
                 task.setOutputDirectory(subproj.file(JAVA_GENERATED_SOURCE_DIRNAME));
                 task.setSource(compileIrTask);
 
                 compileConjure.dependsOn(task);
                 subproj.getTasks().getByName("compileJava").dependsOn(task);
                 applyDependencyForIdeTasks(subproj, task);
                 task.dependsOn(createWriteGitignoreTask(
-                        subproj, "gitignoreConjureRetrofit", subproj.getProjectDir(), JAVA_GITIGNORE_CONTENTS));
+                        subproj, "gitignoreConjure" + upperSuffix, subproj.getProjectDir(), JAVA_GITIGNORE_CONTENTS));
                 task.dependsOn(extractJavaTask);
             });
 
-            ConjureJavaServiceDependencies.configureJavaServiceDependencies(subproj, productDependencyExt);
-            Task cleanTask = project.getTasks().findByName(TASK_CLEAN);
-            cleanTask.dependsOn(project.getTasks().findByName("cleanCompileConjureRetrofit"));
-            subproj.getDependencies().add("api", findDerivedProject(project, objectsProjectName));
-            subproj.getDependencies().add("api", "com.google.guava:guava");
-            subproj.getDependencies().add("api", "com.squareup.retrofit2:retrofit");
-            subproj.getDependencies().add("compileOnly", ANNOTATION_API);
+            Task cleanTask = parentProject.getTasks().findByName(TASK_CLEAN);
+            cleanTask.dependsOn(parentProject.getTasks().findByName("cleanCompileConjure" + upperSuffix));
+            if (isNotObjectsProject) {
+                ConjureJavaServiceDependencies.configureJavaServiceDependencies(subproj, productDependencyExt);
```

Comment:
I think this should only apply to client interfaces: dialogue, jersey, and retrofit. I'm not sure why we ever added it to conjure-undertow, probably bad copy-pasta from the jersey generator.